examples/vision/text_image_denoiser.py

```diff
@@ -0,0 +1,295 @@
+"""
+Title: Text Image Denoiser.
+Author: [Anish B](https://twitter.com/anishhacko)
+Date created: 2021/01/18
+Last modified: 2022/01/27
+Description: Example of Text Image Denoiser coupled with Tesseract OCR engine to improve text readability.
+"""
+"""
+## Introduction
+
+This Example explains a simple **Text Image Denoiser for OCR** using U-Net based
+Architectutre. **Autoencoders** are mostly employed for **Image Restoration** problems
+but in recent times **U-Net** based architectures with **skip-connections** have gained
+popularity for several Image-to-Image tasks. We aim to solve the problem with simple
+**Pretrained U-Net as Encoder** and **Efficient Sub-Pixel CNN as Decoder**. The problem
+we focus here is enhancing the Document Images which are deteriorated through external
+degradations like blur, corrupt text blocks etc. We have followed up on few seminal
+papers which are presented below for reference. At the end of this tutorial user will
+gain clear understanding of building Custom Image Pre-Processors using Deep-Learning that
+helps us to mitigate real world OCR issues.
+The following example requires an additional Installation of of the following packages
+[pybind11](https://github.com/pybind/pybind11),
+[fastwer](https://github.com/kahne/fastwer),
+[pytesseract](https://pypi.org/project/pytesseract/) and
+[tesseract-ocr](https://github.com/tesseract-ocr/tesseract#installing-tesseract).
+Executing **Additional Setup** codeblock should do the job.
+
+
+**References:**
+- [Enhancing OCR Accuracy with Super-Resolution](https://cdn.iiit.ac.in/cdn/cvit.iiit.ac.in/images/ConferencePapers/2018/ocr_Ankit_Lat_ICPR_2018.pdf)
+
+- [Improving the Perceptual Quality of Document Images Using Deep Neural Network](http://mile.ee.iisc.ac.in/publications/softCopy/DocumentAnalysis/ISNN_11page_65.pdf)
+"""
+
+"""
+## Additional Set-up
+"""
+
+# !pip install pybind11
+# !pip install fastwer
+# !pip install pytesseract
+# !sudo apt install tesseract-ocr
+
+import fastwer
+import pytesseract
+import numpy as np
+from glob import glob
+import matplotlib.pyplot as plt
+
+import tensorflow as tf
+from tensorflow import keras
+from tensorflow.keras.applications import resnet_v2
+
+"""
+## Dataset
+We have created a synthetic document dataset that depicts several real world document
+noises; we have used Blurs and Morphological filters to reconstruct real world noises.
```

Comment:
@anish9 
...; we have used blurs and morph... 

It should start with a new sentence, like 

> We have created a synthetic document dataset that depicts several real-world document
noises. We have used blurs and morphological filters to reconstruct real-world noises

Also, you mention you create this dataset. Would you like to mention the data creation process in a few words?

Reply:
@innat, please check this 
A Synthetic Document Dataset that depicts several real-world document noises is used to train the model. The Dataset is injected with Multiple combinations of random Noises with random Kernels to create Blur and Morphological effects to form Noisy Representation.

Reply:
LGTM